### PR TITLE
boards: use hooks instead of SYS_INIT

### PR DESCRIPTION
**Pull request by nashif**
- **board: 96boards: use board hooks instead of SYS_INIT**
- **board: arduino: use board hooks instead of SYS_INIT**
- **board: arm: use board hooks instead of SYS_INIT**
- **board: blues: use board hooks instead of SYS_INIT**
- **board: digilent: use board hooks instead of SYS_INIT**
- **board: nxp: use board hooks instead of SYS_INIT**
- **board: particle: use board hooks instead of SYS_INIT**
- **board: qemu: use board hooks instead of SYS_INIT**
- **board: silabs: use board hooks instead of SYS_INIT**
- **board: snps: use board hooks instead of SYS_INIT**
- **board: sparkfun: use board hooks instead of SYS_INIT**
- **board: st: use board hooks instead of SYS_INIT**

### CODE REVIEW

**Review by decsny — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]
Approved for ST board

File: boards/silabs/dev_kits/sltb010a/Kconfig.defconfig [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Copyright (c) 2023 Antmicro <www.antmicro.com>
 # SPDX-License-Identifier: Apache-2.0
 
+if BOARD_SLTB010A
```
Comment (jerome-pouiller) [inline boards/silabs/dev_kits/sltb010a/Kconfig.defconfig:4]: Shouldn't this change be put in a separate commit?

File: boards/silabs/dev_kits/sltb004a/Kconfig.defconfig [hunk 2]
```diff
@@ -9,4 +9,7 @@ config CMU_HFXO_FREQ
 config CMU_LFXO_FREQ
 	default 32768
 
+config BOARD_LATE_INIT_HOOK
+	def_bool y if CCS811
```
Comment (jerome-pouiller) [inline boards/silabs/dev_kits/sltb004a/Kconfig.defconfig:13]: Since `BOARD_LATE_INIT_HOOK` is defined elsewhere, I believe `def_bool` is not relevant and should be replaced by `default`, no?

File: boards/silabs/dev_kits/sltb010a/Kconfig.defconfig [hunk 3]
```diff
@@ -30,3 +31,7 @@ endif # BT
 
 config REGULATOR
 	default y if SI7210
+
+config BOARD_LATE_INIT_HOOK
+	def_bool y if UART_GECKO
```
Comment (jerome-pouiller) [inline boards/silabs/dev_kits/sltb010a/Kconfig.defconfig:36]: I believe `board_late_init_hook()` need to be called unconditionally. Not only if UARt is enabled.

File: boards/silabs/dev_kits/xg24_dk2601b/board.c [hunk 4]
```diff
@@ -29,15 +29,13 @@ static int efr32xg24_dk2601b_init(void)
 
 
 	if (!gpio_is_ready_dt(&wake_up_gpio_dev)) {
-		LOG_ERR("Wake-up GPIO device was not found!\n");
-		return -ENODEV;
+		LOG_ERR("Wake-up GPIO device was not found!");
```
Comment (jerome-pouiller) [inline boards/silabs/dev_kits/xg24_dk2601b/board.c:32]: Nit pick: the "\n" removal is not related to this work.

File: boards/silabs/dev_kits/xg24_dk2601b/board.c [hunk 5]
```diff
@@ -29,15 +29,13 @@ static int efr32xg24_dk2601b_init(void)
...
 	}
 	ret = gpio_pin_configure_dt(&wake_up_gpio_dev, GPIO_OUTPUT_ACTIVE);
 	if (ret < 0) {
-		return ret;
+		LOG_ERR("Failed to configure wake-up GPIO device!");
 	}
```
Comment (jerome-pouiller) [inline boards/silabs/dev_kits/xg24_dk2601b/board.c:37]: Just curiosity, what is the purpose of configuration the GPIO in board_late_init_hook() rather than using `output-high` in the device tree?
